docs/ibc/middleware/develop.md

```diff
@@ -0,0 +1,230 @@
+
+`Base Application`: A base application is an IBC application that does not contain any middleware. It may be nested by 0 or multiple middleware to form an application stack.
+
+`Application Stack (or stack)`: A stack is the complete set of application logic (middleware(s) +  base application) that gets connected to core IBC. A stack may be just a base application, or it may be a series of middlewares that nest a base application.
+
+## Create a custom IBC Middleware
+
+IBC Middleware will wrap over an underlying IBC application and sits between core IBC and the application. It has complete control in modifying any message coming from IBC to the application, and any message coming from the application to core IBC. Thus, middleware must be completely trusted by chain developers who wish to integrate them, however this gives them complete flexibility in modifying the application(s) they wrap.
+
+#### Interfaces
+
+```typescript
+// Middleware implements the ICS26 Module interface
+interface Middleware extends ICS26Module {
+    app: ICS26Module // middleware has acccess to an underlying application which may be wrapped by more middleware
+    ics4Wrapper: ICS4Wrapper // middleware has access to ICS4Wrapper which may be core IBC Channel Handler or a higher-level middleware that wraps this middleware.
+}
+```
+
+```typescript
+// This is implemented by ICS4 and all middleware that are wrapping base application.
+// The base application will call `sendPacket` or `writeAcknowledgement` of the middleware directly above them
+// which will call the next middleware until it reaches the core IBC handler.
+interface ICS4Wrapper {
+    sendPacket(packet: Packet)
+    writeAcknowledgement(packet: Packet, ack: Acknowledgement)
+}
+```
+
+### Implement `IBCModule` interface and callbacks
+
+The middleware must have access to the underlying application, and be called before during all ICS-26 callbacks. It may execute custom logic during these callbacks, and then call the underlying application's callback. Middleware **may** choose not to call the underlying application's callback at all. Though these should generally be limited to error cases.
+
+In the case where the IBC middleware expects to speak to a compatible IBC middleware on the counterparty chain; they must use the channel handshake to negotiate the middleware version without interfering in the version negotiation of the underlying application.
+
+Middleware accomplishes this by formatting the version in the following format: `{mw-version}:{app-version}`.
+
+During the handshake callbacks, the middleware can split the version into: `mw-version`, `app-version`. It can do its negotiation logic on `mw-version`, and pass the `app-version` to the underlying application.
+
+### Handshake Callbacks:
```

Comment:
I don't see any mention of capability handling?

Reply:
There is no capability handling anymore in middleware. Capabilities are passed down to base app which will claim them

Reply:
> There is no capability handling anymore in middleware. Capabilities are passed down to base app which will claim them

Yes but this needs to be explained. Otherwise a middleware may attempt to claim a capability